### PR TITLE
feat: add flags

### LINKED ISSUE #69: add flags
Add flags, where students can ask for help from the Teacher.

### PR DESCRIPTION
**Pull request by spaenleh**
Flagged comments are not shown for the moment in the teacher interface.

closes #69

### CODE REVIEW

**Comment by spaenleh** [comment]
@juancarlosfarah sorry for the big PR, this adds general reporting flag functionality as well as bot-only "ask for a human" actions. For now only the "ask for human" are shown to the teacher. The flagged items are not displayed in the teacher view.

In this PR i also correct the position of the Modals to show where the action was made (at the right height).

Here an example of the report dialog:
Capture d’écran 2022-02-28 à 15 09 02

**Comment by spaenleh** [comment]
Update: Add a view to show flagged comments

File: src/langs/en.json [hunk 1]
```diff
@@ -68,6 +69,12 @@
     "You can reply with": "You can reply with",
     "Quick Replies": "Quick Replies",
     "Restart interaction": "Restart interaction",
-    "Default Version": "Default Version"
+    "Report": "Report",
+    "Report a comment": "Report a comment",
+    "Default Version": "Default Version",
+    "Add Reaction": "Add Reaction",
+    "Reason": "Reason",
+    "Please provide bellow the reason for reporting this comment": "Please provide bellow the reason for reporting this comment",
```
Comment (juancarlosfarah) [inline src/langs/en.json:77]: `below`

File: src/components/modes/teacher/TeacherView.js [hunk 2]
```diff
@@ -163,20 +209,37 @@ export class TeacherView extends Component {
...
       <>
         <Grid container spacing={0}>
           <Grid item xs={12} className={classes.main}>
             <Typography variant="h6" color="inherit">
               {t('Student Comments')}
             </Typography>
+            <div align="right">
+              <FormControlLabel
+                control={switchComponent}
+                label={t('Need Help Only')}
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js]: Only Show Help Requests

File: src/components/modes/teacher/TeacherView.js [hunk 3]
```diff
@@ -114,9 +121,11 @@ export class TeacherView extends Component {
     const {
       students,
       comments,
+      t,
       dispatchOpenFeedbackView,
       dispatchSetSelectedStudent,
       dispatchGetAppInstanceResources,
+      settings: { helpWantedOnly = DEFAULT_HELP_WANTED_ONLY_SETTING },
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js:128]: `helpWanted` => `helpRequestsOnly`

File: src/components/modes/teacher/AvatarDialog.js [hunk 4]
```diff
@@ -454,6 +466,14 @@ class AvatarDialog extends Component {
                 />
               </Grid>
             ) : null}
+            {avatar.autoBot ? (
+              <Grid item>
+                <FormControlLabel
+                  control={allowHumanInterventionSwitchControl}
+                  label={t('Allow Human Intervention')}
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/AvatarDialog.js:473]: Maybe allow help requests?

Comment (juancarlosfarah) [reply src/components/modes/teacher/AvatarDialog.js:473]: "Allow Request Human Intervention"

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Looks good! Only minor changes!

**Review by juancarlosfarah — APPROVED** [review]
👍

**Pull request merged**